### ISSUE TITLE
Auto-completion for R (R.sublime-completion file)

### ISSUE THREAD

**Issue by jlegewie** [karthik/Rtools#2]
Hi, 
I am not sure how auto completions work for you but I create a R.sublime-completions file at one point. Here is a description:

http://readthedocs.org/docs/sublime-text-unofficial-documentation/en/latest/extensibility/completions.html?highlight=sublime-completions

So the auto completions come from different sources. One is the sublime-completions file. Check out my R.sublime-completions to see whether such a file improves the auto-complete proposals ST2 makes. I can't really tell whether it makes a difference because it's to long ago that I added this file. Here it is (I think you just have to put it in your packages/R folder:

https://gist.github.com/2336626

Note that this file includes functions that are particular to my packages and my own functions. So it will make suggestions that do not make sense for you occasionally. If such the sublime-completions file improves auto-completion for R, I can send you code for the Rtools package that automatically creates a auto-completion based on the currently loaded packages in R. It's not going to be straight forward to integrate this code in the ST2 package but it would be possible to create a R.sublime-completions for the base and some popular R packages or invest some time to integrate the automatic creation of the file in Rtools.

**Comment by karthik** [comment]
I never bothered with auto-completions until now. ST2 seems to be smart about learning variable and function names from a current script and starts to autocomplete after first use. I dropped your gist into my R/ folder in ST2/packages and I like that it now autocompletes base R stuff.

I very much appreciate your contributions and am excited that more people are interested in using ST2 for R development. How do you feel about me just giving you write permission to the master branch of Rtools? I tried to email you but couldn't find one in your github profile.

**Comment by karthik** [comment]
PS: I started a wiki to keep track of new features that we (or other interested folks) could contribute to this repo.

**Comment by jlegewie** [comment]
okay, thanks. 

I personally prefer if feature ideas are issues with a corresponding tag 'feature idea' (or something like that). I think that is the place most people would look first...

**Comment by karthik** [comment]
Makes sense. I remove the pointer to the wiki and ask people to submit issues instead. Also, added you as an admin on the repo.
